### PR TITLE
Collisions in random explicit nonces

### PR DESCRIPTION
**Pull request by martinthomson**
We were already more or less saying this, but let's be really, really
explicit that this is not considered and not recommended.

Wording is not something I'm happy with exactly, so feel free to suggest alternatives.

### CODE REVIEW

File: draft-irtf-cfrg-aead-limits.md [hunk 1]
```diff
@@ -511,11 +511,14 @@ a secret, random offset. The bound for nonce randomization was further improved
...
-nonce is a non-repeating counter. The full nonce is the concatenation of the
-two parts. This produces similar limits under most conditions.
+captured by Theorem 5.3 in {{GCM-MU2}}, which apply to protocols such as TLS 1.2
+{{?RFC5246}}. Here, the implicit part of the nonce is a random value, of length
+at least 32 bits and fixed per key, while we assume that the explicit part of
+the nonce is chosen using a non-repeating process. The full nonce is the
+concatenation of the two parts. This produces similar limits under most
+conditions.  Note that implementations that choose the explicit part at random
+have a higher chance of nonce collisions and are neither recommended nor
+considered.
```
Comment (chris-wood) [inline draft-irtf-cfrg-aead-limits.md:521]:
```suggestion
have a higher chance of nonce collisions and are not considered for the
limits in this section.
```

Comment (chris-wood) [reply draft-irtf-cfrg-aead-limits.md:521]: I would omit the 'not recommended' part, as that's probably something left to TLS 1.2 and its related documents.

**Review by chris-wood — APPROVED** [review]
LGTM pending one suggestion.

**Review by fxguenther — APPROVED** [review]
Looks good, thanks!

**Pull request merged**